### DISCUSSION TITLE
Suggest adding Auth0 as an alternative to Feide OpenIdp identity provider

### DISCUSSION THREAD

**Discussion by law108000** [bergie/passport-saml#769] [Ideas]
As I am unable to register as SP on the [Feide OpenIdp identity provider](https://openidp.feide.no/) by your [instructions](https://github.com/node-saml/passport-saml/blob/master/README.md).

However, I found that [Auth0](https://auth0.com/) is allowed to create an application with SAML easily.
It would be better to mention it on the instruction page for new users to experience your library.

![Screenshot 2022-09-08 at 12 43 51 PM](https://user-images.githubusercontent.com/8112024/189036653-03b773b7-731a-4ce9-bcea-c2b521b1f1a8.png)

No settings are needed to enable SAML.
![Screenshot 2022-09-08 at 12 51 29 PM](https://user-images.githubusercontent.com/8112024/189046318-e5ee8819-210b-4444-a12f-57ce34e4dc03.png)

And all info you need is ready for use.
![Screenshot 2022-09-08 at 2 13 42 PM](https://user-images.githubusercontent.com/8112024/189048091-c91950c7-0826-4c0c-b66e-34abd42d2219.png)

**Comment by srd90** [comment]
Random sidenote:
@law108000 In one of the pictures you have/had  in your post (at the time when this comment was written) you provided a sample configuration which has two issues:
1. `audience` is not verified (you are most probably using `passport-saml <= 3.x.x` which has disabled it by default)
2. you have explicitly disabled `NotOnOrAfter` validation (by setting `acceptedClockSkewMs` to negative value) which means that `AuthnResponse` can be reused (replayed) multiple times (until certificate expires) to login to your application without having to visit IdP.
   * see [v3.2.1/README.md#config-parameter-details](https://github.com/node-saml/passport-saml/blob/v3.2.1/README.md#config-parameter-details)

Additional links relating to accepted clock skew problem:
* https://github.com/node-saml/passport-saml/issues/137#issuecomment-883482011
* https://github.com/node-saml/passport-saml/issues/520#issuecomment-917679604
* https://github.com/node-saml/passport-saml/issues/430#issuecomment-1049217109
* https://github.com/node-saml/passport-saml/discussions/703#discussioncomment-3019381
* #704 

For anyone reading configuration example from one of the images which is attached to first post at the time of writing this particular comment: do not disable `NotOnOrAfter` validation by setting accepted clock skew to negative value (i.e. do not copy paste such configuration from anywhere).

**Reply by law108000** [reply]
Thank you @srd90. 
I already updated `audience` and `acceptedClockSkewMs` to 5000 instead of -1. It works well.
However, I am using `passport-saml@3.2.1` and the `audience` config is not required by default.

My package.json
![Screenshot 2022-09-09 at 1 58 06 PM](https://user-images.githubusercontent.com/8112024/189281412-7e99c402-53eb-4c95-8400-d2e9e39e8b99.png)
 
Latest configuration
![Screenshot 2022-09-09 at 3 51 12 PM](https://user-images.githubusercontent.com/8112024/189300353-2ac99dc3-efc3-4490-8407-d5ea6b2d3e32.png)

**Reply by markstos** [reply]
I would welcome a patch to the documentation to suggest Auth0 is an alternative. SAML is already hard enough to work as it is. I'm sure others would welcome a recommendation of an easy-to-use demo. 

Or maybe this documentation should be moved to a wiki page where it's easier for people to make edits, as SAML providers will continue to change over time.

**Reply by cjbarth** [reply]
@markstos I like the idea of a wiki page instead of README in the repo. Do you think we should move what we have to a wiki page and point people there to update it as things evolve with products and the codebase?
